src/class.cr

```diff
@@ -8,10 +8,92 @@ class Class
     hasher.class(self)
   end
 
-  def ==(other : Class)
+  # Returns whether this class is the same as *other*.
+  #
+  # ```
+  # Int32 == Int32  # => true
+  # Int32 == String # => false
+  # ```
+  def ==(other : Class) : Bool
     crystal_type_id == other.crystal_type_id
   end
 
+  # Returns whether this class inherits or includes *other*.
+  #
+  # ```
+  # Int32 < Number  # => true
+  # Int32 < Value   # => true
+  # Int32 < Int32   # => false
+  # Int32 <= String # => false
+  # ```
+  def <(other : T.class) : Bool forall T
+    # This is so that the method is expanded differently for each type
+    {% @type %}
+    other._gt(self)
+  end
+
+  # Returns whether this class inherits or includes *other*, or
+  # is equal to *other*.
+  #
+  # ```
+  # Int32 < Number  # => true
+  # Int32 < Value   # => true
+  # Int32 <= Int32  # => true
+  # Int32 <= String # => false
+  # ```
+  def <=(other : T.class) : Bool forall T
+    # This is so that the method is expanded differently for each type
+    {% @type %}
+    other._gte(self)
+  end
+
+  # Returns whether *other* inherits or includes `self`.
+  #
+  # ```
+  # Number > Int32  # => true
+  # Number > Number # => false
+  # Number > Object # => false
+  # ```
+  def >(other : T.class) forall T
+    # This is so that the method is expanded differently for each type
+    {% @type %}
+    other._lt(self)
+  end
+
+  # Returns whether *other* inherits or includes `self`, or is equal
+  # to `self`.
+  #
+  # ```
+  # Number >= Int32  # => true
+  # Number >= Number # => true
+  # Number >= Object # => false
+  # ```
+  def >=(other : T.class) forall T
+    # This is so that the method is expanded differently for each type
+    {% @type %}
+    other._lte(self)
+  end
+
+  # :nodoc:
+  def _lt(other : T.class) forall T
```

Comment:
Why not make it `protected` (and all below)?

Reply:
I tried, but it doesn't work. The compiler doesn't like it for some reason and I have no time/will to investigate it now.